plugins/auto-sizes/tests/improve-sizes-test.php

```diff
@@ -8,32 +8,54 @@
 
 class Tests_Improve_Sizes extends WP_UnitTestCase {
 
+	/**
+	 * Attachment ID.
+	 *
+	 * @var int
+	 */
+	public static $image_id;
+
 	/**
 	 * Set up the environment for the tests.
 	 */
-	public function setUp(): void {
-		parent::setUp();
+	public static function set_up_before_class(): void {
+		parent::set_up_before_class();
 
 		switch_theme( 'twentytwentyfour' );
 	}
 
+	/**
+	 * Setup shared fixtures.
+	 */
+	public static function wpSetUpBeforeClass( WP_UnitTest_Factory $factory ): void {
+		self::$image_id = $factory->attachment->create_upload_object( TESTS_PLUGIN_DIR . '/tests/data/images/leaves.jpg' );
+	}
```

Comment:
Nice! I think this can be combined with the `set_up_before_class()` method above, but you'll need to call `self::factory()->attachment->{etc)`.

Reply:
Fixed in https://github.com/WordPress/performance/pull/1250/commits/d866befe667846653cd5eff35e36f21e19d65979